src/memfd.rs

```diff
@@ -210,7 +210,7 @@ impl Memfd {
         let fd = self.file.as_raw_fd();
         let flags = sealing::seals_to_bitflags(seals);
         // UNSAFE(lucab): required syscall.
-        let r = unsafe { libc::syscall(libc::SYS_fcntl, fd, libc::F_ADD_SEALS, flags) };
+        let r = unsafe { libc::fcntl(fd, libc::F_ADD_SEALS, flags) };
```

Comment:
I recall there being a reason to use the `syscall` interface. Not sure what it was, however.

Reply:
Ok. Do you remember? Using `libc::fcntl` feel more portable than the explicit sys call name. I replaced `syscall` with `fnctl` because I had build errors for `aarch64-linux-android` with the `SYS_fcntl` arg which is not exposed by libc. I haven't digged which syscall name to use instead which would also be an option with a `cfg(target_os = ...` guard.

Reply:
It looks like I wrote it that way initially, but I don't remember exactly why. Trying to retro-guess myself, it's likely one of those reasons:
 * `fcntl` is a muxed wrapper in most libc implementations, and I didn't see a real reason to go through an additional opaque layer (and to make writing seccomp filters easier)
 * I was possibly playing with this on a platform where libc-rs was missing the function binding (uclibc? musl?) and so I went for the syscall directly
 * I was playing with the idea of making this work just with [sc](https://docs.rs/sc) and without libc-rs at all, but I never walked that path till the end

Or possibly a mix of all of them. The only interesting one nowadays is the first, but portability is likely a better argument.

Reply:
@lucab Thanks for your reply. Which way would you like to go? I can try to refactor this patch to use `libc::syscall` with`target_os` (and maybe `target_arch`) specific guards. Otherwise we can also check which targets are affected at all.

Reply:
@flxo if you are not in a hurry, it may be worth to just dig a bit more into libc-rs. If it is just missing a few `SYS_fcntl` definitions, it would be good to add it there; they have a self-service release process so it shouldn't add to much delay before being able to consume it here.
Instead, if the reason turns out to be more complex than that and we don't find any blocker for switching to the libc muxed wrapper, we can freely do that as a fallback.
In any case, thanks for looking into this :)

Reply:
@lucab I'll have a look why my first attempt with the direct use of `libc::syscall` didn't work for `aarch64-linux-android`.

Reply:
@lucab Turned out that `SYS_fcntl` is missing in `aarch64-linux-android` in libc. I opened [#2308](https://github.com/rust-lang/libc/pull/2308)

The flags `F_ADD_SEALS` and `F_GET_SEALS` are already included in libc for this target.
I'd propose to wait for the PR unless you're fine with the use of `libc::fcntl`...